packages/datetime/test/datePickerTests.tsx

```diff
@@ -26,6 +30,139 @@ describe("<DatePicker>", () => {
         assert.isUndefined(root.state("selectedDay"));
     });
 
+    describe("reconciliates dayPickerProps", () => {
+        it("week starts with firstDayOfWeek value", () => {
+            const selectedFirstDay = 3;
+            const wrapper = mount(<DatePicker dayPickerProps={{ firstDayOfWeek: selectedFirstDay }} />);
+            const firstWeekday = wrapper.find("Weekday").first();
+            assert.equal(firstWeekday.prop("weekday"), selectedFirstDay);
+        });
+
+        it("doesnt show outside days if specified", () => {
+            const defaultValue = new Date(2017, Months.SEPTEMBER, 1, 12);
+            const wrapper = mount(
+                <DatePicker defaultValue={defaultValue} dayPickerProps={{ enableOutsideDays: false }} />,
+            );
+
+            const dayIsHidden = (day: any): boolean =>
+                day.prop("empty") && !day.prop("ariaSelected") && day.prop("ariaDisabled");
+
+            assert.isTrue(dayIsHidden(wrapper.find("Day").at(0)));
+            assert.isTrue(dayIsHidden(wrapper.find("Day").at(1)));
+            assert.isTrue(dayIsHidden(wrapper.find("Day").at(2)));
+            assert.isTrue(dayIsHidden(wrapper.find("Day").at(3)));
+            assert.isTrue(dayIsHidden(wrapper.find("Day").at(4)));
+
+            assert.isFalse(dayIsHidden(wrapper.find("Day").at(5)));
+        });
+
+        it("respects default of true for enableOutsideDays prop", () => {
+            const defaultValue = new Date(2017, Months.SEPTEMBER, 1);
+            const firstDayInView = new Date(2017, Months.AUGUST, 27, 12, 0);
+            const wrapper = mount(<DatePicker defaultValue={defaultValue} />);
+            const firstDay = wrapper.find("Day").first();
+            assert.isTrue(areDatesEqual(new Date(firstDay.prop("day")), firstDayInView));
```

Comment:
Can we just do `assertDatesEqual` if we use this in more than one place?

```tsx
function assertDatesEqual(date1: Date, date2: Date) {
    assert.isTrue(areDatesEqual(date1, date2));
}
```

or just put all the logic in `assertDatesEqual`.